src/ignore.rs

```diff
@@ -14,14 +15,19 @@ pub fn project_dirs() -> ProjectDirs {
         .expect("Could not find project directory.")
 }
 
-#[derive(Debug)]
 pub struct GitIgnore {
     server: String,
     cache_dir: PathBuf,
     ignore_file: PathBuf,
+    detectors: Vec<Detector>,
     pub config: Option<Config>,
 }
 
+struct Detector {
```

Comment:
This would probably benefit from having a `Detector::new` and could probably be in its own file :+1: